test/functional/feature_index_prune.py

```diff
@@ -19,9 +19,23 @@ def set_test_params(self):
             ["-fastprune", "-prune=1", "-blockfilterindex=1"],
             ["-fastprune", "-prune=1", "-coinstatsindex=1"],
             ["-fastprune", "-prune=1", "-blockfilterindex=1", "-coinstatsindex=1"],
-            []
+            [],
         ]
 
+    def setup_network(self):
+        self.setup_nodes()  # No P2P connection, so that linear_sync works
+
+    def linear_sync(self, node_from, *, height_from=None):
+        # Linear sync over RPC, because P2P sync may not be linear
+        to_height = node_from.getblockcount()
+        for n in self.nodes:
+            for i in range(height_from or n.getblockcount(), to_height + 1):
```

Comment:
could start at `n.getblockcount() + 1`, nodes will already have the block at `n.getblockcount()`.